// @tailwindcss/postcss/dist/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});// src/index.ts
require('postcss');
var _tailwindcss = require('tailwindcss');
var _oxide = require('@tailwindcss/oxide');
function tailwindcss(opts) {
  return {
    postcssPlugin: "tailwindcss-v4",
    AtRule: {
      tailwind: async (atRule, { result, postcss }) => {
        console.time("Tailwind CSS");
        console.time("Resolve project");
        let project = _oxide.resolveProject.call(void 0, { base: process.cwd() });
        console.timeEnd("Resolve project");
        console.time("Compile CSS");
        let css = _tailwindcss.preflight + _tailwindcss.compile.call(void 0, project.candidates);
        console.timeEnd("Compile CSS");
        for (let file of project.files) {
          result.messages.push({
            type: "dependency",
            plugin: "tailwindcss-v4",
            file,
            parent: result.opts.from
          });
        }
        for (let { base, glob } of project.globs) {
          result.messages.push({
            type: "dir-dependency",
            plugin: "tailwindcss-v4",
            dir: base,
            glob,
            parent: result.opts.from
          });
        }
        console.time("Parsing CSS");
        let result2 = postcss.parse(css);
        console.timeEnd("Parsing CSS");
        atRule.replaceWith(result2);
        console.timeEnd("Tailwind CSS");
      }
    }
  };
}
var src_default = Object.assign(tailwindcss, { postcss: true });


exports.default = src_default;

module.exports = exports.default;
